homeassistant/components/vizio/config_flow.py

```diff
@@ -338,28 +336,25 @@ async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
         return await self.async_step_user(user_input=import_config)
 
     async def async_step_zeroconf(
-        self, discovery_info: DiscoveryInfoType | None = None
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
     ) -> FlowResult:
         """Handle zeroconf discovery."""
+        host = discovery_info[zeroconf.ATTR_HOST]
         # If host already has port, no need to add it again
-        if ":" not in discovery_info[CONF_HOST]:
-            discovery_info[
-                CONF_HOST
-            ] = f"{discovery_info[CONF_HOST]}:{discovery_info[CONF_PORT]}"
+        if ":" not in host:
+            host = f"{host}:{discovery_info[zeroconf.ATTR_PORT]}"
```

Comment:
Note: I have explicitely extracted to discovery_info into three separate variables (host/name/device_class) to avoid confusion with the original ZeroconfServiceInfo which should really be considered immutable in my opinion.